### DISCUSSION TITLE
New number format for durations

### DISCUSSION THREAD

**Discussion by avicenteg** [apache/superset#18401] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I'm working with durations, Superset offers me by default two options:

![image](https://user-images.githubusercontent.com/64475023/94570951-b074c480-026f-11eb-8506-bf44862e7e4c.png)

The first one is my favourite but sometimes I would like to have a simpler format. Instead of _HH:mm:ss.s_ I'd prefer _HH:mm_.

**Describe the solution you'd like**
Add a new format in the format number section for durations coming from ms. 

Duration in ms (66000 => 1m)

**Describe alternatives you've considered**

I think that my proposed solution is easy adding this option to the number-format package.

[a comment by an automated account is not shown]

**Comment by villebro** [comment]
@avicenteg I'd be interested in hearing more about your use case. We could potentially add the `colonNotation` option (`HH:mm:ss` as opposed to the traditional `H m s` format) that [pretty-ms](https://github.com/sindresorhus/pretty-ms) supports as a default duration formatting option. You can check the unit tests there to see which format suits you needs best. In the long term the plan is to make it possible to pass the options to the formatter, much like is done for D3 now.

**Comment by avicenteg** [comment]
@villebro In fact, I'm more interested in not having the seconds part. Maybe use the colonNotation isn't a good idea since it may lead to confussion with Time fields. 
I think that the option `compact` in pretty-ms is the best way to get it, but I'm not sure if it will delete also the minutes. 

For example:

3660000 ms => 1 h 57 m 

It will be great to have an option of passing the options to the formatter, but I understand that it is a little bit trickier than for D3.

**Comment by jon-coffey** [comment]
Na idea how to achive this? I'm also only interessted in the days and maybe hours of the milliseconds.

Is there any other way how to format numbers e.g. to inject a custom number formatter to superset without recompling the frontend?

**Comment by jcarlos92** [comment]
Are we still considering support for this functionality? We're exploring the possibility of implementing something similar and would appreciate any updates.

**Comment by villebro** [comment]
Yeah, this would be great. I think what would be needed is moving away from these static formatters, and introducing a notion of formatter plugins, which you can then pass configs to. For `d3`, it would accept whichever formatting parameters that accepts, for `pretty-ms` the typical configs for that etc. I don't think this will be a massive change, but I'd propose opening a SIP for it, just to make sure everyone is able to comment on the final design before it gets implemented.
